hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1138,7 +1120,11 @@ public DirectoryInfo(String relativePath, FileStatus[] fileStatus) {
           this.isHoodiePartition = true;
         } else if (FSUtils.isDataFile(status.getPath())) {
           // Regular HUDI data file (base file or log file)
-          filenameToSizeMap.put(status.getPath().getName(), status.getLen());
+          String dataFileCommitTime = FSUtils.getCommitTime(status.getPath().getName());
```

Comment:
Incase of MOR table, for a log file, the base instance time could be < actual delta commit time. So, we might skip the log files based on this logic? 
since in L1125, we are filtering for files whose commit time < max instant time. May be we should use last mod time instead?

Reply:
may be this bug exists even if not for this patch. we might need a follow up fix.